streams/src/main/java/org/apache/kafka/streams/processor/internals/GlobalStateManagerImpl.java

```diff
@@ -275,31 +259,70 @@ private void restoreState(final StateRestoreCallback stateRestoreCallback,
                               final RecordConverter recordConverter) {
         for (final TopicPartition topicPartition : topicPartitions) {
             globalConsumer.assign(Collections.singletonList(topicPartition));
+            long offset;
             final Long checkpoint = checkpointFileCache.get(topicPartition);
             if (checkpoint != null) {
                 globalConsumer.seek(topicPartition, checkpoint);
+                offset = checkpoint;
```

Comment:
Some side improvement: if we `seek` base on the checkpoint, there is no reason to call `position()` because we know what our offset is. -- Only if we `seekToBeginning()` we need to get the current offset from the consumer itself (instead of `position()` we could also call `beginningOffsets` but `position` it the easer to use API.